swap/src/protocol/bob/state.rs

```diff
@@ -347,13 +347,15 @@ impl State2 {
                 punish_timelock: self.punish_timelock,
                 refund_address: self.refund_address,
                 redeem_address: self.redeem_address,
+                punish_address: self.punish_address,
```

Comment:
https://github.com/comit-network/xmr-btc-swap/blob/master/swap/src/network/swap_setup/alice.rs#L69

Does it mean that punish_address equals redeem_address?

Reply:
> We want to avoid changes to the state machine types because this is a breaking change. It's not backwards compatible with swaps that were started before this change.
> 
> Is there a way to do this differently?

I think that's the easiest way to do this. Can you notify users about the breaking change in release notes?
Or we can read the punish_address from db in the refund function.

Reply:
Well, actually, I can implement check_for_tx_punish on state3, then load it from db in fn refund.

Reply:
I can only change State2 and then load it from the db in the refund function to minimize the changes to the db.